### DISCUSSION TITLE
Smooth Transition between Animations

### DISCUSSION THREAD

**Discussion by smessmer** [bevyengine/bevy#6338] [Q&A] [answered]
I'm loading a skeleton mesh from a glTF file and it has several animations defined (idle, running, walking, attacking, ...). If using `AnimationPlayer.play()`, it is a very hard transition between one and the next animation, this looks weird. I would like to smoothly transition between those animations if the character, say, starts or stops running. What is the best way of doing this? Is there a way to smoothly transition animations?

**Comment by james7132** [comment, marked as answer]
Right now there isn't a decent implementation of this provided by Bevy natively. https://github.com/bevyengine/rfcs/pull/49 and https://github.com/bevyengine/rfcs/pull/51 provide the framework for building these kinds of systems. The design is meant to allow generic blending, not just transitions, so it would provide the framework for macro and micro-level control over more complex animation systems, but you may need to build your own knobs on top of a lower level API to make it work. 

For example, a transition between two animations would be a weighted blend of two clips:

 - The source animation where we start at a weight of 1.0 that progressively trends towards 0.0.
 - The destination animation where we start at a weight of 0.0 that progressively trends towards 1.0.
 -  As time progresses, we advance both clips at the same time rate while shifting the blend towards the destination.
